xonsh/proc.py

```diff
@@ -1008,9 +1009,6 @@ def run(self):
         if self.f is None:
             return
         last_in_pipeline = self._wait_and_getattr('last_in_pipeline')
-        if last_in_pipeline:
-            capout = self._wait_and_getattr('captured_stdout')
-            caperr = self._wait_and_getattr('captured_stderr')
```

Comment:
These shouldn't be removed per se, because they guarantee that the pipeline can read from stdout & stderr file pipes. Also, I think that this has been slightly refactored in one of my other PRs.